File: Scripts/js/FN_extend.js
// @grant nodejs
console.log("⏳ 初始化安装依赖中......")
$exec('wget https://raw.githubusercontent.com/Oreomeow/freenom-py/main/requirements.txt -O requirements.txt && pip3 install -r requirements.txt', {
  cwd: './script/Shell',
  timeout: 0,
  cb(data, error) {
    error ? console.error(error) : console.log(data)
  }
})
console.log("⏳ 开始拉取 git 仓库 Oreomeow/freenom-py")
$exec('git clone https://github.com/Oreomeow/freenom-py.git', {
  cwd: './script/Shell',
  timeout: 0,
  cb(data, error) {
    error ? console.error(error) : console.log(data)
  }
})
console.log("⏳ 开始执行 FN_extend.py")
$exec('python3 FN_extend.py', {
  cwd: './script/Shell/freenom-py',
  timeout: 0,
  env: {
    FN_ID: $store.get('FN_ID', 'string'),
    FN_PW: $store.get('FN_PW', 'string'),
    MAIL_USER: $store.get('MAIL_USER', 'string'),
    MAIL_ADDRESS: $store.get('MAIL_ADDRESS', 'string'),
    MAIL_PW: $store.get('MAIL_PW', 'string'),
    MAIL_HOST: $store.get('MAIL_HOST', 'string'),
    MAIL_PORT: $store.get('MAIL_PORT', 'string'),
    MAIL_TO: $store.get('MAIL_TO', 'string'),
  },
  cb(data, error) {
    error ? console.error(error) : console.log(data)
  }
})

File: Scripts/py/FreeNom/FN_extend.py
# -*- coding: utf8 -*-
'''
Author: shuai93
Modifier: Oreo
Date: Wed Aug 11 10:15:41 UTC 2021
cron: 25 7 */10 * *
new Env('Freenom 续期邮件版');
------------
环境变量说明                     示例
FN_ID: Freenom 用户名           1234567890@gmail.com
FN_PW: Freenom 密码             12345678
MAIL_USER: 发件人邮箱用户名      address@vip.qq.com 或 123456@qq.com
MAIL_ADDRESS: 发件人邮箱地址     address@vip.qq.com 或 123456@qq.com
MAIL_PW: 发件人邮箱授权码        xxxxxxxxxxxxxxxx 看下方链接
* MAIL_HOST: 发件人邮箱服务器    smtp.qq.com 不填默认为这个
* MAIL_PORT: 邮箱服务器端口      465 不填默认为这个
MAIL_TO: 收件人邮箱可与发件人相同 address@vip.qq.com 或 123456@qq.com   

填写总参考：https://service.mail.qq.com/cgi-bin/help?subtype=1&&id=28&&no=369
------------
依赖模块说明
pip install -r requirements.txt / pip3 install -r requirements.txt
'''
from utils.settings import *
from utils.exception import CustomException
from utils.freenom import FreeNom
from utils.mail import EmailPoster


def main():
    print("配置信息")
    print([MAIL_TO, MAIL_PORT, MAIL_HOST, MAIL_ADDRESS, MAIL_PW, MAIL_USER, FN_ID, FN_PW])
    if not all([MAIL_TO, MAIL_PORT, MAIL_HOST, MAIL_ADDRESS, MAIL_PW, MAIL_USER, FN_ID, FN_PW]):
        raise CustomException("参数缺失")

    to = [MAIL_TO]

    body = {
        'subject': "FreeNom 自动续期",
        'to': to,
    }
    try:
        results = FreeNom().run()
        body['payload'] = {
            "results": results,
            "user": FN_ID
        }
    except CustomException as e:
        body['body'] = e.message

    EmailPoster().send(data=body)


if __name__ == "__main__":
    main()
